ksqldb-common/src/test/java/io/confluent/ksql/util/KsqlVersionTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.junit.Test;
+
+public class KsqlVersionTest {
+
+  @Test
+  public void shouldOnlyCompareMajorMinorVersionsInSame() {
+    assertThat(new KsqlVersion("5.1.1").same(new KsqlVersion("5.1.0")), is(true));
+    assertThat(new KsqlVersion("5.1.1").same(new KsqlVersion("5.1.1")), is(true));
+    assertThat(new KsqlVersion("5.1.1").same(new KsqlVersion("5.1.2")), is(true));
+
+    assertThat(new KsqlVersion("5.1.0").same(new KsqlVersion("5.0.0")), is(false));
+    assertThat(new KsqlVersion("5.1.0").same(new KsqlVersion("5.2.0")), is(false));
+    assertThat(new KsqlVersion("5.1.0").same(new KsqlVersion("6.1.0")), is(false));
+    assertThat(new KsqlVersion("5.1.0").same(new KsqlVersion("6.1.0")), is(false));
+  }
+
+  @Test
+  public void shouldCompareCpVersionToStandaloneVersionInSame() {
+    // known mappings
+    assertThat(new KsqlVersion("6.0.").same(new KsqlVersion("0.10.")), is(true));
+    assertThat(new KsqlVersion("0.10.").same(new KsqlVersion("6.0.")), is(true));
+
+    assertThat(new KsqlVersion("6.1.").same(new KsqlVersion("0.14.")), is(true));
+    assertThat(new KsqlVersion("0.14.").same(new KsqlVersion("6.1.")), is(true));
+
```

Comment:
Add tests for known mapping that compare to false?

Same for the `isAtLeast()` tests below.